kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ExecutionContext.kt

```diff
@@ -71,13 +71,19 @@ interface ErrorCode {
  * @see ErrorCode
  */
 fun errorCode(code: Int) = object : ErrorCode {
-    override val value = code
+    override val value: Int = code
+
+    override fun equals(other: Any?): Boolean = this === other || (other is ErrorCode && value == other.value)
+
+    override fun hashCode(): Int = value.hashCode()
+
+    override fun toString(): String = value.toString()
 }
 
-open class ErrorResult(
-    open val code: ErrorCode = ErrorCode.UNDEFINED,
-    open val message: String? = null,
-    open val throwable: Throwable? = null
+data class ErrorResult(
```

Comment:
Not really sure that `ErrorResult` should be final. In the CSOB app we extend it to modify code variable type and add some additional values.

Reply:
Why is @smidf  final? This class is extend it very often. Specific process has own error result implementation based on this base class

Reply:
Yes, I think the `ErrorResult` extending is anti-pattern. The `Error` class / state is something, that describes the error, and the code (unique identifier), the message and the stacktrace are sufficient. If a user has the ability to extend, add, or override class attributes, he can exploit the error to pass data.

Only `ErrorCode` should be domain specific, not `ErrorResult`.